File: server/Models/VaultKeep.cs
using System.ComponentModel.DataAnnotations;
using keepr.interfaces;

namespace keepr.Models;

public class VaultKeep : IRepoItem<int>
{
    public int Id { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public int KeepId { get; set; }
    public int VaultId { get; set; }
    public string CreatorId { get; set; }
    public Keep Keep { get; set; }
}

public class VaultKeepCreationDTO
{
    public int KeepId { get; set; }
    public int VaultId { get; set; }
    public int VaultKeepId { get; set; }
}

File: client/src/models/VaultKeep.js
import { Profile } from "./Profile.js"

export class VaultKeep{
    constructor(data){
        this.id = data.id
        this.keepId = data.keepId
        this.vaultId = data.vaultId
        this.creatorId = data.creatorId
        this.createdAt = new Date(data.createdAt)
        this.updatedAt = new Date(data.updatedAt)
        this.creator = data.creator ? new Profile(data.creator) : null
    }
}